state-sync/src/coordinator.rs

```diff
@@ -199,69 +198,116 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
                 (network_id, event) = network_events.select_next_some() => {
                     match event {
                         Event::NewPeer(peer_id, origin) => {
-                            let peer = PeerNetworkId(network_id, peer_id);
-                            self.request_manager.enable_peer(peer, origin);
-                            self.check_progress();
+                            if let Err(e) = self.process_new_peer(network_id, peer_id, origin) {
+                                error!(LogSchema::new(LogEntry::NewPeer).error(&e.into()));
+                            }
                         }
                         Event::LostPeer(peer_id, origin) => {
-                            let peer = PeerNetworkId(network_id, peer_id);
-                            self.request_manager.disable_peer(&peer, origin);
+                            if let Err(e) = self.process_lost_peer(network_id, peer_id, origin) {
+                                error!(LogSchema::new(LogEntry::LostPeer).error(&e.into()));
+                            }
+                        }
+                        Event::Message(peer_id, message) => {
+                            if let Err(e) = self.process_chunk_message(network_id.clone(), peer_id, message).await {
+                                error!(LogSchema::new(LogEntry::ProcessChunkMessage).error(&e.into()));
+                            }
                         }
-                        Event::Message(peer_id, message) => self.process_one_message(PeerNetworkId(network_id.clone(), peer_id), message).await,
                         unexpected_event => {
                             counters::NETWORK_ERROR_COUNT.inc();
                             warn!(LogSchema::new(LogEntry::NetworkError),
                             "received unexpected network event: {:?}", unexpected_event);
                         },
+
                     }
                 },
                 _ = interval.select_next_some() => {
-                    self.check_progress();
+                    match self.check_progress() {
+                        Ok(()) => {},
+                        Err(Error::ConsensusIsExecuting) => { /* Don't log to avoid spamming logs! */ },
+                        Err(error) => {
+                            error!(LogSchema::event_log(LogEntry::ProgressCheck, LogEvent::Fail).error(&error.into()));
+                        }
+                    }
```

Comment:
Perhaps 1 less line, but if this makes more sense to you stick with it
```
if let Err(error) = self.check_progress() {
  if !matches(error, Error::ConsensusIsExecuting) {
    error!(...)
  }
}
```

Reply:
Now that you're saying `ConsensusIsExecuting` is essentially an unimportant error, were there others that could just be ignored?  Would metrics based on these errors be a better approach?

Reply:
We don't want to log "ConsensusIsExecuting" here because this check_progress() code branch is executed every 100ms (see `config.tick_interval_ms`) using an interval stream, i.e., it's not a malicious case: it's possible that consensus might have taken control of the node, but state sync will continue to execute/handle these messages at the same time and thus spam the logs.